### PR TITLE
Use Webhook Fields in WFJT Form

### LINKED ISSUE #6742: Use Webhook Fields in WFJT Form
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY
We just created a webhook fields subform for JT Form.  That same component could also be used for WFJT form.  

The webhook key and the webhook url field need some changes mentioned here https://github.com/ansible/awx/pull/6654#pullrequestreview-395653664

### PR DESCRIPTION
**Pull request by nixocio**
Update WorkflowJobTemplateForm to use `WebhookSubForm`. Also, update
related unit-tests.

closes: https://github.com/ansible/awx/issues/6742

### CODE REVIEW

File: awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx [hunk 1]
```diff
@@ -512,7 +512,7 @@ function JobTemplateForm({
                         &nbsp;
                         <FieldTooltip
                           content={i18n._(
-                            t`Enable webhook for this workflow job template.`
+                            t`Enable webhook for this job template.`
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx]: ~I think we should dynamically change this string based on the template type to be `Enable webhook for this workflow job template` or `Enable webhook for this job template`~

File: awx/ui_next/src/screens/Template/shared/WorkflowJobTemplateForm.jsx [hunk 2]
```diff
@@ -60,125 +46,12 @@ function WorkflowJobTemplateForm({
...
-      setContentError(err);
-    }
-  };
-
-  if (hasContentError || contentError) {
-    return <ContentError error={contentError || hasContentError} />;
-  }
-
-  if (contentLoading) {
-    return <ContentLoading />;
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Template/shared/WorkflowJobTemplateForm.jsx:181]: So happy to see all this code removed. Way to go!

File: awx/ui_next/src/screens/Template/shared/WorkflowJobTemplateForm.test.jsx [hunk 3]
```diff
@@ -183,31 +184,32 @@ describe('<WorkflowJobTemplateForm/>', () => {
...
     ).toBe(true);
     expect(
       wrapper.find('input[aria-label="wfjt-webhook-key"]').prop('value')
     ).toBe('sdfghjklmnbvcdsew435678iokjhgfd');
     await act(() =>
-      wrapper
-        .find('FormGroup[name="webhook_key"]')
-        .find('Button[variant="tertiary"]')
-        .prop('onClick')()
+      wrapper.find('Button[aria-label="Update webhook key"]').prop('onClick')()
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Template/shared/WorkflowJobTemplateForm.test.jsx:194]: Better selector, good job!

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
~So there are some changes requested here. More details are in the comments.  Looks like there are 2 failing tests for me locally in WebhooksSubForm.test.jsx.~ ( I failed to rebase correctly.  Tests aren't failing)  Good work on this.

File: awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx (hunk 1 above)
Comment (jlmitch5) [reply awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx]: I'm also fine if we just make it "for this template", as we tend to use that word to mean JT or WFJT

File: awx/ui_next/src/screens/Template/shared/WebhookSubForm.jsx [hunk 4]
```diff
@@ -85,9 +90,12 @@ function WebhookSubForm({ i18n, enableWebhooks }) {
     useCallback(async () => {
       const {
         data: { webhook_key: key },
-      } = await JobTemplatesAPI.updateWebhookKey(id);
+      } =
+        templateType === 'job_template'
```
Comment (jlmitch5) [inline awx/ui_next/src/screens/Template/shared/WebhookSubForm.jsx]: seems a little weird having await in the ternary like this, I'd probably set the updateWebhookKey in a var above and then just call that here.  in other words

const updateWebhookKey = templateType === 'job_template' ? JobTemplatesAPI.updateWebhookKey : WorkflowJobTemplatesAPI.updateWebhookKey

await updateWebhookKey(id);

File: awx/ui_next/src/screens/Template/shared/WebhookSubForm.test.jsx [hunk 5]
```diff
@@ -99,7 +100,7 @@ describe('<WebhooksSubForm />', () => {
               webhook_key: 'A NEW WEBHOOK KEY WILL BE GENERATED ON SAVE.',
             }}
           >
-            <WebhookSubForm enableWebhooks />
+            <WebhookSubForm enableWebhooks templateType="job_template" />
           </Formik>
         </Route>,
         {
```
Comment (jlmitch5) [inline awx/ui_next/src/screens/Template/shared/WebhookSubForm.test.jsx:106]: might be good to have a test for the workflow templateType too

**Review by jlmitch5 — COMMENTED** [review]
looks good so far just some small things

File: awx/ui_next/src/screens/Template/shared/WebhookSubForm.test.jsx (hunk 5 above)
Comment (AlexSCorey) [reply awx/ui_next/src/screens/Template/shared/WebhookSubForm.test.jsx:106]: Yeah I agree with @jlmitch5. You could also test that the webhook url contains the proper template type.

File: awx/ui_next/src/screens/Template/shared/WebhookSubForm.jsx (hunk 4 above)
Comment (nixocio) [reply awx/ui_next/src/screens/Template/shared/WebhookSubForm.jsx]: Good suggestion.

**Review by jlmitch5 — APPROVED** [review]
looks good!

**Review by AlexSCorey — APPROVED** [review]
Looks good to me.  Way to go on removing so much code in the Workflow Job Template Form!  That file will be way more maintainable now.

**Review by dsesami — CHANGES_REQUESTED** [review]
404 occurs if you try to refresh the webhook key while editing the form.
1) make a webhook on a new WFJT
2) save the WFJT
3) edit the WFJT and try to press the refresh button on the webhook key

**Comment by nixocio** [comment]
Thanks @dsesami, I will take a look.

**Comment by nixocio** [comment]
Fixed @dsesami.

**Review by dsesami — APPROVED** [review]

**Pull request merged**